### PR TITLE
Move HOC event review out from under Pegasus teacher dashboard

### PR DESCRIPTION
**Pull request by Erin007**
[LP-2155](https://codedotorg.atlassian.net/browse/LP-2155)

Long overdue clean up to move HoC Event Review from the Pegasus teacher dashboard, which was moved to Code Studio in spring of 2019 (example: #27868). Given that there is no longer a teacher dashboard on Pegasus, it doesn't make sense to continue to include teacher-dashboard in the url for HoC Event Review. 

**BEFORE:** https://code.org/teacher-dashboard/review-hociyskvuwa 

**AFTER**: https://code.org/review-hociyskvuwa

The functionality of the HoC Review pages is un-changed.  Hitting https://code.org/teacher-dashboard/review-hociyskvuwa  now redirects to https://code.org/review-hociyskvuwa.

### CODE REVIEW

**Comment by Erin007** [comment]
I think this is incremental progress toward a cleaner Pegasus that's worth doing, but it's brought up two questions for me: 

1.) Do we have a philosophy for how long we keep redirects around? I'm in favor of the best user experience and understand some may have bookmarked links, but wonder if there's a time tipping point at which the long term code cruft impacts ease of maintainability? I skimmed a few articles ([example](https://www.highervisibility.com/blog/how-long-keep-301-redirect-before-removing/)) and it seems like at least a year is recommended. Maybe I'm just letting Marie Kondo carry me away. :) 

2.) Is Code Studio actually a better long-term home for this HOC Event Review content? The functionality of the page depends on authentication, particularly that you are a teacher, and if we want to keep Code.org/Pegasus as user-agnostic as possible with the exception of the header, maybe this content shouldn't live on Pegasus?

File: cookbooks/cdo-varnish/libraries/http_cache.rb [hunk 1]
```diff
@@ -139,6 +139,7 @@ def self.config(env)
               /amazon-future-engineer*
               /create-company-profile*
               /edit-company-profile*
+              /review-hociyskvuwa*
```
Comment (maureensturgeon) [inline cookbooks/cdo-varnish/libraries/http_cache.rb:142]: For my learning, could you explain why this change is necessary?

**Review by maureensturgeon — APPROVED** [review]

**Review by jamescodeorg — APPROVED** [review]
I kind of feel like this shouldn't be at the top-level, but I see that we have other things at the top-level in that list in http_cache.rb that look dubious so :shrug:.  Could we open a ticket to see if we should add some stronger permissioning to this?  Looks like we're currently relying on security-by-obscurity.  😬 

And thanks for cleaning this up!  :-)

File: cookbooks/cdo-varnish/libraries/http_cache.rb (hunk 1 above)
Comment (Erin007) [reply cookbooks/cdo-varnish/libraries/http_cache.rb:142]: This is necessary because reviewing HOC events requires authentication - we need to know who you are and that you're a teacher. Adding the url to this list allows that information to be available on the Pegasus page.

**Comment by Erin007** [comment]
> I kind of feel like this shouldn't be at the top-level, but I see that we have other things at the top-level in that list in http_cache.rb that look dubious so 🤷. Could we open a ticket to see if we should add some stronger permissioning to this? Looks like we're currently relying on security-by-obscurity. 😬
> 
> And thanks for cleaning this up! :-)

Added a new Jira ticket here: https://codedotorg.atlassian.net/browse/LP-2175 Stronger permissions here and a broader clean up of the outdated cruft on Pegasus would be ideal. Thanks for raising!
